.github/workflows/test-and-publish.yaml

```diff
@@ -49,6 +49,11 @@ jobs:
           echo $(poetry run python --version)
           poetry build --format=wheel
           ls -lah ./dist/
+      - name: Upload wheel as CI artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheel-${{ github.run_id }}-${{ github.sha }}
```

Comment:
run_id and github.sha don't change between matrices

I suggest adding both `-{{ matrix.os }}-{{ matrix.python_version }}` to the end of the wheel.